### PR TITLE
Abstract client stream id and use a logical id

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

- Use a logical id based on streamType and metadata
- Abstract ClientStream id

## Related issues

related to #12402

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@npepinpe I have only cleaned up the PR. So it is mostly your code. If you would like to do a review pass, please do. Otherwise, I will approve and merge this PR.

File: transport/src/main/java/io/camunda/zeebe/transport/stream/api/ClientStreamId.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.transport.stream.api;
+
+/** ID of a registered client stream */
+public interface ClientStreamId {}
```
Comment (npepinpe) [inline transport/src/main/java/io/camunda/zeebe/transport/stream/api/ClientStreamId.java:11]: :star_struck: Love the use of an opaque ID

**Comment by npepinpe** [comment]
Thanks for making me look better than I am ;) 

I trust you, and I can't officially review my own PR anyway, so go ahead and merge it when you want. It would be great to have a test that ensures removing works with whatever ID is provided by add, but I guess we don't really have those tests just yet (i.e. tests for the registry entry point). Those are more integration tests, I suppose. 

Speaking of, maybe we should add integration tests for the registry (with all its sub components). Since this is our entry point for the ZPA team, it feels strange that it's not tested at all. But we can add them as part of https://github.com/camunda/zeebe/issues/12388

**Comment by deepthidevaki** [comment]
> Speaking of, maybe we should add integration tests for the registry (with all its sub components).

There is one test StreamIntegationTest https://github.com/camunda/zeebe/blob/0cf51450cfbfc045a132b652fa2c49fc4c16e92c/transport/src/test/java/io/camunda/zeebe/transport/stream/impl/StreamIntegrationTest.java#L156 which test for removed client stream. 
Do you mean some other case?

**Review by deepthidevaki — APPROVED** [review]
bors merge

**Pull request merged**